conans/model/options.py

```diff
@@ -214,6 +218,13 @@ def update_options(self, other, is_pattern=False):
                 continue
             self._set(k, v)
 
+    def constrain_possible_values(self, field, possible_values):
+        self._ensure_exists(field)
+        if self._freeze:
+            raise ConanException(f"Cannot constrain option '{field}' possible values after it has "
+                                 f"been set to '{self._data[field].value}'")
+        self._data[field]._constrain_possible_values(possible_values)
```

Comment:
A bit ugly, but we can't make this private as it would not be freeze-aware calling it from `self.options.myoption...`